components/layout_2020/flow/inline.rs

```diff
@@ -184,12 +193,144 @@ impl LineUnderConstruction {
     }
 }
 
+/// A block size relative to a line's final baseline. This is to track the size
+/// contribution of a particular element of a line above and below the baseline.
+/// These sizes can be combined with other baseline relative sizes before the
+/// final baseline position is known. The values here are relative to the
+/// overall line's baseline and *not* the nested baseline of an inline box.
+#[derive(Clone, Debug)]
+struct BaselineRelativeSize {
+    /// The ascent above the baseline, where a positive value means a larger
+    /// ascent. Thus, the top of this size contribution is `baseline_offset -
+    /// ascent`.
+    ascent: Au,
+
+    /// The descent below the baseline, where a positive value means a larger
+    /// descent. Thus, the bottom of this size contribution is `baseline_offset +
+    /// descent`.
+    descent: Au,
+}
+
+impl BaselineRelativeSize {
+    fn zero() -> Self {
+        Self {
+            ascent: Au::zero(),
+            descent: Au::zero(),
+        }
+    }
+
+    fn max(&self, other: &Self) -> Self {
+        BaselineRelativeSize {
+            ascent: self.ascent.max(other.ascent),
+            descent: self.descent.max(other.descent),
+        }
+    }
+
+    /// Given a nested baseline offset, ie the offset of the baseline of a child
+    /// inline box relative to the line's baseline, adjust this
+    /// [`BaselineRelativeSize`] so that it is relative to the line's baseline..
```

Comment:
```
// Given an offset from the line's overall baseline, adjust this [`BaselineRelativeSize`]
// by that offset. This is used to adjust a [`BaselineRelativeSize`] for different kinds
// of baseline-relative `vertical-align`. This will "move" measured size of a particular
// inline box's block size. For example, in the following HTML:
```

Reply:
I've updated this documentation to be clearer with the text above and after consultation with @mukilan .